### PR TITLE
docs: Make goals and non-goals clearer

### PR DESCRIPTION
**Pull request by riddell-stan**
Make httpstan's goals and non-goals clearer.

### CODE REVIEW

File: doc/contributing.rst [hunk 1]
```diff
@@ -7,23 +7,36 @@ call functions in the Stan C++ library's ``stan::services`` namespace.
 **httpstan** was originally developed as a "backend" for a Stan interface
 written in Python, PyStan_.
 
-httpstan emerged out of a rewrite of an existing piece of software (PyStan)
-which was difficult to maintain. httpstan aims to be maintainable. Code
-contributions should be readable and easy to understand.
+Goals:
+
+- Provide access to frequently-used functions in Stan C++ library's ``stan::services`` namespace.
```
Comment (ahartikainen) [inline doc/contributing.rst:12]: Just a comment/question, no need to change anything.


`Provide access to frequently-used functions`

I like the wording here, but why we don't try to add access to all functions in `stan::services`?

File: doc/contributing.rst [hunk 2]
```diff
@@ -7,23 +7,36 @@ call functions in the Stan C++ library's ``stan::services`` namespace.
...
+- Maintainability.
+
+Non-goals:
+
+- Provide access to functions other than those in the ``stan::services`` namespace.
+  There are only three exceptions: ``model_base_crtp.log_prob``,
+  ``model_base_crtp.write_array``, and ``stan::model::log_prob_grad``.
+
+If these goals and non-goals strike you as restrictive, remember that httpstan
+is open source software which you are free to fork and customize.
```
Comment (ahartikainen) [inline doc/contributing.rst:22]: Is there a way to soften the tone of this sentence.

There might be a valid case for other functionality, and we just want kindly to remind users that people are free to fork and customize package if needed.

File: doc/contributing.rst [hunk 3]
```diff
@@ -7,23 +7,36 @@ call functions in the Stan C++ library's ``stan::services`` namespace.
...
 
 .. _shim: https://en.wikipedia.org/wiki/Shim_%28computing%29
 .. _PyStan: http://mc-stan.org/interfaces/pystan.html
 
 How to Make a Code Contribution
 ===============================
 
+Code contributions must be readable and easy to understand.
+httpstan emerged out of a rewrite of an existing piece of software (PyStan 2)
+which was difficult to maintain. httpstan aims to be maintainable.
```
Comment (ahartikainen) [inline doc/contributing.rst:32]: Softer tone here too.

I'm not 100% how to express this in english, but something like `this in mind, httpstan aims ...`

**Review by ahartikainen — APPROVED** [review]
I like the definitions.

I commented a couple of locations where the language could be a bit more conversational (I wrote softer tone, but probably I mean conversational). That being said, I might have misinterpret the tone of the message, so nothing to be worried about.

File: doc/contributing.rst (hunk 1 above)
Comment (ahartikainen) [reply doc/contributing.rst:12]: Yep, read what is in there,  `frequently-used` is good.

**Comment by riddell-stan** [comment]
@ahartikainen Thanks for the comments. I'll make some revisions this weekend.

**Pull request merged**